Spec/Auth.swift

```diff
@@ -185,6 +185,67 @@ class Auth : QuickSpec {
                         expect(client.auth.method).to(equal(ARTAuthMethod.Token))
                     }
                 }
+
+                // RSA4b
+                it("in REST, if the token creation failed or the subsequent request with the new token failed due to a token error, then the request should result in an error") {
+                    let options = AblyTests.clientOptions()
+                    options.key = AblyTests.commonAppSetup().key
+                    options.token = getTestToken()
+
+                    let rest = ARTRest(options: options)
+                    rest.httpExecutor = testHTTPExecutor
+
+                    let channel = rest.channels.get("test")
+
+                    testHTTPExecutor.afterRequest = { _ in
+                        testHTTPExecutor.simulateIncomingServerErrorOnNextRequest(40141, description: "token revoked")
+                    }
+
+                    testHTTPExecutor.simulateIncomingServerErrorOnNextRequest(40141, description: "token revoked")
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.publish("message", data: nil) { error in
+                            guard let error = error else {
+                                fail("Error is nil"); done(); return
+                            }
+                            expect(error.code).to(equal(40141))
+                            done()
+                        }
+                    }
+                }
+
+                // RSA4b
+                it("in Realtime, if the token creation failed or the subsequent request with the new token failed due to a token error, then the connection should move to the FAILED state") {
+                    let options = AblyTests.clientOptions()
+                    options.key = AblyTests.commonAppSetup().key
+                    options.token = getTestToken(ttl: 0.5)
```

Comment:
As above, providing a token explicitly is not a means to renew (authCallback or authUrl should be used ideally to demonstrate the point)